DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/lldp/linux/linux_lldp_impl.py

```diff
@@ -70,3 +70,74 @@ def parse_set(self, command, output, *argv, **kwarg):
         # TODO: Implement me
 
         return cmd
+
+    def format_configure(self, command, *argv, **kwarg):
+        """
+        Usage:   lldpcli [OPTIONS ...] [COMMAND ...]
+        Version: lldpd 2020-09-23
+        -d          Enable more debugging information.
+        -u socket   Specify the Unix-domain socket used for communication with lldpd(8).
+        -f format   Choose output format (plain, keyvalue, json, json0, xml).
+        see manual page lldpcli(8) for more information
+        """
+        params = kwarg['params']
+        cmd = 'lldpcli {} '.format(command)
+        # custom code here
+        if 'interface' in params:
+            cmd += 'ports {} '.format(params.get('interface'))
+        if 'lldp' in params:
+            cmd += 'lldp '
+        if 'status' in params:
+            cmd += 'status {} '.format(params.get('status'))
+        if 'tx-interval' in params:
+            cmd += 'tx-interval {} '.format(params.get('tx-interval'))
+        if 'tx-hold' in params:
+            cmd += 'tx-hold {} '.format(params.get('tx-hold'))
+        if 'system' in params:
+            cmd += 'system {} '.format(params.get('system'))
+        if 'real' in params:
+            cmd += 'real '
+        return cmd
+
+    def format_show_lldpcli(self, command, *argv, **kwarg):
+        """
+        Usage:   lldpcli [OPTIONS ...] [COMMAND ...]
+        Version: lldpd 2020-09-23
+        -d          Enable more debugging information.
+        -u socket   Specify the Unix-domain socket used for communication with lldpd(8).
+        -f format   Choose output format (plain, keyvalue, json, json0, xml).
+        see manual page lldpcli(8) for more information
+        """
+        params = kwarg['params']
+        command = command.split('_')[0]
+        cmd = 'lldpcli {} {} '.format(params.get('cmd_options', ''), command)
+        if 'neighbors' in params:
+            cmd += 'neighbors '
+        if 'statistics' in params:
+            cmd += 'statistics '
+        if 'interfaces' in params:
+            cmd += 'interfaces '
+        if 'ports' in params:
+            cmd += 'ports '
+        if 'chassis' in params:
+            cmd += 'chassis '
+        if 'interface' in params:
+            cmd += '{} '.format(params.get('interface'))
+        if 'running-configuration' in params:
+            cmd += 'running-configuration '
+        return cmd
+
+    def parse_show_lldpcli(self, command, output, *argv, **kwarg):
+        """
+        Usage:   lldpcli [OPTIONS ...] [COMMAND ...]
+        Version: lldpd 2020-09-23
+        -d          Enable more debugging information.
+        -u socket   Specify the Unix-domain socket used for communication with lldpd(8).
+        -f format   Choose output format (plain, keyvalue, json, json0, xml).
+        see manual page lldpcli(8) for more information
+        """
+        try:
+            parsed_out = json.loads(output)
+        except json.decoder.JSONDecodeError:
+            return []
```

Comment:
If we don't provide "-j" option to cmd_options but the result will be valid, it will return nothing. Better to return base output in such case

Reply:
It's not better to return the base output here. The parsed output will go into `out[0][dent]['parsed_output']` and the user will expect there to be a dict or a list (not a base output string).
If returning an empty list is not ok here then it is better to not handle the exception at all.

Reply:
Let's just return json.loads(output) here, since both the empty list and plain output may be misleading for the user

Reply:
Done